docs/source/Examples/lambci.rst

```diff
@@ -0,0 +1,303 @@
+	      # path to the lambci directory - the following sub directories will be used:
+	      # * task    directory of lambda function, /var/task in container
+	      # * runtime directory of the lambda runtime, /opt in container
+	      # * layers  directory of layer pool with each layer a sub directory
+	      # * env     environment file for lambci docker container
+	      directory : '~/lambci'
+	      # address of lambci server API
+	      address : 'localhost:9001'
+
+*******************************************************************************
+Install the function
+*******************************************************************************
+
+Install the ``echo`` example function using SRK with the following command:
+
+::
+
+	$ ./srk --config configs/lambci-local.yaml function create -s examples/echo
+
+
+*******************************************************************************
+Start the LambCI server
+*******************************************************************************
+
+SRK does not presently start the LambCI FaaS provider on its own, so you will
+need to start it manually. The following starts up a container
+running a LambCI server locally:
+
+::
+
+	LAMBCI_PATH=$HOME/lambci
+	LAMBCI_RUNTIME=python3.8
+	LAMBCI_HANDLER=lambda_function.lambda_handler
+	docker run --rm -d \
+	  --name srk-lambci \
+	  -v $LAMBCI_PATH/task:/var/task:ro,delegated \
+	  -v $LAMBCI_PATH/runtime:/opt:ro,delegated \
+	  --env-file $LAMBCI_PATH/env \
+	  -e DOCKER_LAMBDA_STAY_OPEN=1 \
+	  -p 9001:9001 \
+	  lambci/lambda:$LAMBCI_RUNTIME \
+	  $LAMBCI_HANDLER
+
+You may also want to create a script to streamline this process.
+See `Scripting LambCI Startup`_.
+
+*******************************************************************************
+Invoke the function
+*******************************************************************************
+
+Use SRK to invoke the ``echo`` function using the following command:
+
+::
+
+	./srk --config configs/lambci-local.yaml bench -b one-shot -a '{"hello" : "world"}'
+
+
+*******************************************************************************
+Shut down the LambCI server
+*******************************************************************************
+
+::
+
+	docker kill srk-lambci
+
+*******************************************************************************
+Using Custom Libraries
+*******************************************************************************
+
+SRK with LambCI allows you to add custom libraries to the runtime. The
+configuration procedure parallels that of AWS Lambda, which users
+`layers <https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html>`_.
+
+In this example we will create a runtime that includes Python's popular `Requests <https://requests.readthedocs.io/en/master/>`_.
+Use the commands below to install the layer:
+
+::
+
+	mkdir -p $HOME/lambci/layers/requests-layer/python
+	pip3 install requests -t $HOME/lambci/layers/requests-layer/python
```

Comment:
Actually --files as currently implemented is a quirk to get some additional files into the task (lambda function) directory while layers seem to be the more general way to do this. Layers can be configured beforehand and stacked together in "runtimes" as needed. Also --files wants a list of single files which is not practically for larger packages (though this could be changed).
Maybe we should think about an AWS-independent term / mechanism for "layers" and later on remove the --files option.